### PR TITLE
[BE] typing for decorators - optim/optimizer

### PR DESCRIPTION
**Pull request by aorenste**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131590
* #131589
* #131588
* #131587
* #131586
* #131585
* __->__ #131583
* #131582
* #131581
* #131580
* #131579
* #131578
* #131577
* #131576
* #131575
* #131574
* #131573
* #131572
* #131571
* #131570
* #131569
* #131568

See #131429

### CODE REVIEW

File: torch/optim/optimizer.py [hunk 1]
```diff
@@ -35,6 +36,9 @@
 )
 from torch.utils.hooks import RemovableHandle
 
+_T = TypeVar("_T")
```
Comment (janeyx99) [inline torch/optim/optimizer.py:39]: There's a T for TypeVar("T") below -- what does the underscore mean here?

**Review by janeyx99 — APPROVED** [review]
Looks fine, thanks!

Left a q about the underscore

File: torch/optim/optimizer.py (hunk 1 above)
Comment (aorenste) [reply torch/optim/optimizer.py:39]: Underscore means the name is private to the file. I could have used the 'T' version as well - but I didn't want to make any implied guarantees about the type name having any external meaning.

**Pull request closed without merging**